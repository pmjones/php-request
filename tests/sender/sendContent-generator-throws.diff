001+ Fatal error: Uncaught Error: Call to undefined method SapiResponse::getHeaderCallback() in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php:20
002+ Stack trace:
001- foo
002- failure
003+ #0 /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php(11): SapiResponseSender->registerHeaderCallback(Object(SapiResponse))
004+ #1 /Users/pmjones/Code/pmjones/php-request/tests/sender/sendContent-generator-throws.php(9): SapiResponseSender->send(Object(SapiResponse))
005+ #2 {main}
006+   thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php on line 20