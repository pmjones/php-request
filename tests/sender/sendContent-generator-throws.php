<?php
$response = new SapiResponse();
$response->setContent(function () {
    yield "foo\n";
    throw new RuntimeException("failure");
    yield "bar\n";
});
try {
    (new SapiResponseSender())->send($response);
} catch (RuntimeException $e) {
    echo $e->getMessage();
}
