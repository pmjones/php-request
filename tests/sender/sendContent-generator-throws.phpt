--TEST--
SapiResponseSender::sendContent with generator that throws
--CGI--
--INI--
expose_php=0
--FILE--
<?php
$response = new SapiResponse();
$response->setContent(function () {
    yield "foo\n";
    throw new RuntimeException("failure");
    yield "bar\n";
});
try {
    (new SapiResponseSender())->send($response);
} catch (RuntimeException $e) {
    echo $e->getMessage();
}
--EXPECT--
foo
failure
