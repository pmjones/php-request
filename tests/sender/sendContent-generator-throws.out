Fatal error: Uncaught Error: Call to undefined method SapiResponse::getHeaderCallback() in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php:20
Stack trace:
#0 /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php(11): SapiResponseSender->registerHeaderCallback(Object(SapiResponse))
#1 /Users/pmjones/Code/pmjones/php-request/tests/sender/sendContent-generator-throws.php(9): SapiResponseSender->send(Object(SapiResponse))
#2 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php on line 20